commit 13198b133fadc88f6b030e762ae0581dd04b374f
Author: Sage Weil <sweil@redhat.com>
Date:   Tue Nov 5 11:13:35 2019 -0600

    mgr/MgrClient: fix open condition fix
    
    Fix fc60989bf7a72c35b8f6b8fec2407b3080ad9bbd, which reversed this
    condition.
    
    Signed-off-by: Sage Weil <sage@redhat.com>

diff --git a/src/mgr/MgrClient.cc b/src/mgr/MgrClient.cc
index 0f24ab8171e..976d297e639 100644
--- a/src/mgr/MgrClient.cc
+++ b/src/mgr/MgrClient.cc
@@ -574,7 +574,7 @@ int MgrClient::service_daemon_register(
   daemon_dirty_status = true;
 
   // late register?
-  if (msgr->get_mytype() != CEPH_ENTITY_TYPE_CLIENT && session && session->con) {
+  if (msgr->get_mytype() == CEPH_ENTITY_TYPE_CLIENT && session && session->con) {
     _send_open();
   }
 
